OverlayPlugin.Core/NetworkProcessors/LineMapEffect.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace RainbowMage.OverlayPlugin.NetworkProcessors
+{
+    internal static class MapEffectOpcodes
+    {
+        public const uint LogFileLineID = 257;
+
+        public const ushort OPCODE_2022_05_27 = 0x2E0;
+        public const ushort SIZE_2022_05_27 = 0x0B;
+
+        public const ushort OPCODE_2022_06_21 = 0x27B;
+        public const ushort SIZE_2022_06_21 = 0x0B;
+
+        public const ushort OPCODE_2022_07_08 = 0x7B;
+        public const ushort SIZE_2022_07_08 = 0x0B;
+
+        public const ushort OPCODE_2022_08_17 = 0x195;
+        public const ushort SIZE_2022_08_17 = 0x0B;
+
+        public const ushort OPCODE_2022_08_20 = 0x9A;
+        public const ushort SIZE_2022_08_20 = 0x0B;
+
+        public const ushort OPCODE_2022_08_25 = 0x24A;

```

Comment:
This is thinking out loud, but it might be nice to have an externalized json file of opcodes.  It's easier to update than code and would also let OverlayPlugin cache/download the latest version of the opcodes without requiring a release (with some extra infra).  I don't know that it needs to happen in this PR, but it might be a cleaner way to organize if we have more network data opcode lines.  I certainly can imagine attachments (a8s/p1s) and CEs (rather than having cactbot plugin do that).

Alternatively, there are other opcode resources we could consider depending on too, like https://github.com/karashiiro/FFXIVOpcodes/blob/master/opcodes.json too) where it looks like `EnvironmentControl` is the opcode for MapEvent.  But maybe that dependency being abandoned/broken/wrong is extra risk we don't want.   (This list should also give you the other versions you need.)

Reply:
I think a JSON file is fine for pulling these from but I'm a bit leery on a direct dependency outside the project.

Also I believe that the version number reported by the FFXIV_ACT_Plugin is the version info from the `.ver` file which doesn't line up with `6.05` from that file, which is also not correct since the opcodes in the file are for `6.2`...

Realistically this should probably be a `Dictionary<string, OpcodeInfo>` or something so that the reported version from FFXIV_ACT_Plugin can be mapped directly rather than relying on a switch/case.

Reply:
Yeah, that was my same worry, re: external dependency.  But, I think having this as data and not code feels like a better direction for the future.  Mostly having some thoughts re: obviating what #3172 was trying to do too, but at the OverlayPlugin level and for all opcodes.

Reply:
Yeah, I think that the goal first should be to land #49, and while that's being cleaned up and merged I can convert this to read from json instead?

Reply:
```diff
+        public const ushort OPCODE_2022_09_13 = 0x336;
+        public const ushort SIZE_2022_09_13 = 0x0B;
```

Reply:
Yeah, I have these locally but going to put them in the JSON file instead of pushing another update for them here.